Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -143,51 +203,447 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
+               static_cast<double>(octagon_points[index_of_closest_octagon_point].x);
+  double slope_of_octagon_line = rise / run;
+  double y_intercept_of_octagon_line =
+      slope_of_octagon_line * (0 - octagon_points[index_of_closest_octagon_point].x) +
+      octagon_points[index_of_closest_octagon_point].y;
+
+  rise = static_cast<double>(center_of_screen.y) - static_cast<double>(mouse_point.y);
+  run = static_cast<double>(center_of_screen.x) - static_cast<double>(mouse_point.x);
+  double slope_of_line_from_center_to_mouse = rise / run;
+  if (abs(run) < 1 /*Magic number is range around 0 where we just say the slope is zero*/)
+  {
+    if (mouse_point.y < center_of_screen.y)
+    {
+      mouse_point = octagon_points[NORTH];
+    }
+    if (mouse_point.y > center_of_screen.y)
+    {
+      mouse_point = octagon_points[SOUTH];
+    }
+    return;
+  }
+  double y_intercept_of_line_from_center_to_mouse =
+      slope_of_line_from_center_to_mouse * (0 - center_of_screen.x) + center_of_screen.y;
+
+  POINT point_of_intersection = {0};
+
+  point_of_intersection.x = static_cast<long>(
+      round(((y_intercept_of_octagon_line - y_intercept_of_line_from_center_to_mouse) /
+             (slope_of_line_from_center_to_mouse - slope_of_octagon_line))));
+  point_of_intersection.y = static_cast<long>(
+      round((slope_of_octagon_line * point_of_intersection.x) + y_intercept_of_octagon_line));
+
+  if (Calculate_Distance_between_Points(
+          mouse_point, octagon_points[index_of_closest_octagon_point]) < snapping_distance)
+  {
+    point_of_intersection = octagon_points[index_of_closest_octagon_point];
+  }
+
+  mouse_point = point_of_intersection;
+}
+
+void KeyboardMouse::Lock_Mouse_In_Jail(POINT& mouse_point)
+{
+  // Sage 3/30/2022: Locks the mouse into an octagon (like the case on a real controller)
+  //                The plan to make the octagonal locking happen is to rotate the point so
+  //                four octagon edges mark the top, bottom, and sides of the octagon
+  //                (try to make one of the flat edges on a real controller the bottom instead
+  //                of the notch). Then the octagon is just a square with its corners cut off.
+  //                So, we generate a square and cut the corners of the square off in order
+  //                to make the octagon. Then we check if the point is in those corners that were
+  //                cut off and if it is,we move it towardsthe center until it is inside the octagon.
+
+  POINT octagon_points[8] = {0};
+
+  Generate_Octagon_Points(octagon_points);
+
+  if (Point_Is_Inside_Octagon(mouse_point, octagon_points))
+  {
+    return;
+  }
+  else
+  {
+    Move_Mouse_Point_Along_Gate(mouse_point, octagon_points);
+  }
+
+  /* //Sage 3 / 30 / 2022 : Locks the mouse into a square
+  //Sage 4/3/2022: Please do not remove this code. It is an earlier version of the mouse jail
```

Comment:
This should still be removed. If it ever needs to be changed, then it can be introduced with the change that enables it. Keeping code around that isn't used makes bitrot easier

Reply:
Okay :(